libraries/lib-wx-init/SettingsWX.cpp

```diff
@@ -0,0 +1,226 @@
+#include "SettingsWX.h"
+
+#include <wx/confbase.h>
+#include <wx/fileconf.h>
+
+
+void SettingsWX::DoBeginGroup(const wxString& prefix)
+{
+   if(prefix.StartsWith("/"))
+      mGroupStack.push_back(prefix);
+   else
+   {
+      if(mGroupStack.size() > 1)
+         mGroupStack.push_back(mGroupStack.Last() + "/" + prefix);
+      else
+         mGroupStack.push_back("/" + prefix);
+   }
+   //This creates group if it didn't exist
+   mConfig->SetPath(mGroupStack.Last());
+}
+
+void SettingsWX::DoEndGroup() noexcept
+{
+   assert(mGroupStack.size() > 1);// "No matching DoBeginGroup"
+
+   if(mGroupStack.size() > 1)
+      mGroupStack.pop_back();
+
+   mConfig->SetPath(mGroupStack.Last());
+}
+
+SettingsWX::SettingsWX()
+   : mConfig{wxConfigBase::Get()}
+   , mIsConfigOwner {false}
+{
+   mGroupStack.push_back("/");
+}
+
+SettingsWX::SettingsWX(std::unique_ptr<wxConfigBase> config)
+   : mConfig{std::move(config)}, mIsConfigOwner{true}
+{
+   mGroupStack.push_back("/");
+}
+
+SettingsWX::SettingsWX(const wxString& filepath)
+   : mIsConfigOwner {true}
+{
+   mConfig = std::make_unique<wxFileConfig>(wxEmptyString, wxEmptyString, filepath);
+   mGroupStack.push_back("/");
+}
+
+SettingsWX::~SettingsWX()
+{
+   mConfig->Flush();
+   if(!mIsConfigOwner)
```

Comment:
This scares me a lot, as I don't see how it can be safe. I think this is the case where shared_ptr shines.